pkg/agent/apiserver/handlers/ovsflows/handler.go

```diff
@@ -201,6 +206,7 @@ func HandleFunc(aq agentquerier.AgentQuerier) http.HandlerFunc {
 		pod := r.URL.Query().Get("pod")
 		service := r.URL.Query().Get("service")
 		networkPolicy := r.URL.Query().Get("networkpolicy")
+		policyType := r.URL.Query().Get("type")
```

Comment:
It seems that the previous behavior was to return all flows regardless of NP type. Could we preserve this default behavior instead of making `policyType` _required_,  or did that behavior not make sense to begin with?

Reply:
Personally I feel the behavior does not make sense. If you look at the sample outputs above, there's no mechanism to group the dumps by policy should the query return more than one policy. So simply concat dumped flows for ACNP named "allow-all-monitoring" and ANP named "allow-all-monitoring" as if they are from one policy feels really weird to me.
What I can do though: make the `policyType` not required, only return http.BadRequest when there's ambiguity and prompt user to specify a type. Does that sound good?

Reply:
Yes I agree that it's confusing and not helpful. Especially with some policy types sharing the same OVS tables.

> What I can do though: make the policyType not required, only return http.BadRequest when there's ambiguity and prompt user to specify a type. Does that sound good?

That sounds good to me if it is simple to implement. It's not a very unusual solution I think. I'll let @tnqn comment.

Reply:
Just realized that for K8sNP and ANNP, without the policy type, it is hard to validate that a namespace needs to be specified with the policy name, and thus missing the opportunity to return a BadRequest earlier. @tnqn Do you think it is fine to omit the namespace check initially as well? Or we should enforce policy type as input instead